romdisk/uart.asm

```diff
@@ -144,48 +144,140 @@ uartsnd_main:
         ; uartrcv main routine.
         ; It currently accepts a single parameter: size to receive.
         ; It will try to open the driver registered as SER0.
-        ; TODO: Implement this command to actually save data in a file.
         ; Parameters:
         ;       HL - ARGV
         ;       BC - ARGC
         ; Returns:
         ;       A - 0 on success
+        ; Modifies:
+        ;       IX, DE
         PUBLIC uartrcv_main
 uartrcv_main:
+        ; Set up stack frame
+        ld ix, 0
+        add ix, sp
+
         dec bc  ; Do not count the command itself
+
+        ; Check that size is present
         ld a, b
         or c
-        ret z
+        jp z, _rcv_usage
+        ; Advance past command 
         inc hl
         inc hl
+
+        ; Load stack frame
+        push hl ; remaining argv (ix-2)
```

Comment:
You can actually continue parsing the parameters here since HL and BC already contain argv and argc. For example you can put the parse size in HL and the file name in BC (or NULL if not provided). As such, you don't need to stack frame anymore.

Reply:
Will make updated pull request.